packages/x-data-grid/src/hooks/features/virtualization/useGridVirtualization.tsx

```diff
@@ -61,9 +67,20 @@ export function useGridVirtualization(
     }));
   };
 
+  const setRowVirtualization = (enabled: boolean) => {
+    apiRef.current.setState((state) => ({
+      ...state,
+      virtualization: {
+        ...state.virtualization,
+        enabledForRows: enabled,
```

Comment:
To be aligned with the constraint applied in initializer and other setter?
```suggestion
        enabledForRows: enabled && !props.autoHeight,
```